src/server/request.rs

```diff
@@ -66,17 +68,31 @@ impl<'a, T> Request<'a, T> {
     #[inline]
     pub fn version(&self) -> &HttpVersion { &self.version }
 
-    /*
     /// The target path of this Request.
     #[inline]
-    pub fn path(&self) -> Option<&str> {
-        match *self.uri {
-            RequestUri::AbsolutePath(ref s) => Some(s),
-            RequestUri::AbsoluteUri(ref url) => Some(&url[::url::Position::BeforePath..]),
-            _ => None
+    pub fn path(&self) -> Option<Cow<str>> {
+        match self.uri {
+            RequestUri::AbsolutePath(ref s) => {
+                let url = Url::parse("http://example.com").ok().and_then(|u| u.join(s).ok());
```

Comment:
I was imagine using something more primitive:

```rust
let end = s.find('?').unwrap_or(s.len());
&s[..end]
```

Is there a reason that this is incorrect?